### PR TITLE
Multi site example

### PR DESCRIPTION
**Pull request by cobusc**
Added support for multi-site configurations. The OIDC Settings for each site is managed by a new `OidcSettings` model that was introduced.

Out of the box, the Mozilla OIDC Client does not support multiple sites. In order to facilitate this I had to subclass a few classes. Most notably:
* `OIDCAuthenticationBackend` replaced by `GirlEffectOIDCBackend` in `girleffect_oidc_integration/auth.py`
* `SessionRefresh` middleware replaced by `CustomSessionRefresh` in `wagtail_client/middleware.py` (I'll make a ticket upstream to try and make this simpler.)
* `OIDCAuthenticationRequestView` replaced by `CustomAuthenticationRequestView` in `wagtail_client/views.py`
* `OIDCAuthenticationCallbackView` replaced by `CustomAuthenticationCallbackView` in `wagtail_client/views.py`

### CODE REVIEW

File: wagtail_client/settings.py [hunk 1]
```diff
@@ -219,3 +217,8 @@
         },
     },
 }
+
+
+@property
+def TEST_ME():
```
Comment (AltusBarry) [inline wagtail_client/settings.py]: Can possibly remove this.

File: wagtail_client/views.py [hunk 2]
```diff
@@ -67,8 +69,73 @@ def get_redirect_url(self, *args, **kwargs):
...
-        params.update({"insert": "custom", "parameters": "here"})
+        site = get_current_site(request)
+        if not hasattr(site, "oidcsettings"):
+            raise RuntimeError(f"Site {site} has no settings configured.")
+
+        params.update(site.oidcsettings.extra_params)
         return params
+
+
+class CustomAuthenticationCallbackView(OIDCAuthenticationCallbackView):
```
Comment (AltusBarry) [inline wagtail_client/views.py]: I'm not sure I fully understand what needed changing here.

File: wagtail_client/settings.py (hunk 1 above)
Comment (cobusc) [reply wagtail_client/settings.py]: Oh my...thanks!

File: wagtail_client/views.py (hunk 2 above)
Comment (cobusc) [reply wagtail_client/views.py]: @AltusBarry It is left for demonstrative purposes, in case we need to override these URLs at some point.

**Comment by AltusBarry** [comment]
+1

**Pull request merged**